CondTools/RunInfo/src/LHCInfoPerFillPopConSourceHandler.cc

```diff
@@ -0,0 +1,746 @@
+      endSampleTime = executionTime;
+      nextFillSearchTimestamp = executionTimeIov;
+    } else {
+      edm::LogInfo(m_name) << "Found fill " << lhcFill << " created at " << cond::time::to_boost(startFillTime)
+                           << " ending at " << cond::time::to_boost(endFillTime);
+      endSampleTime = cond::time::to_boost(endFillTime);
+      nextFillSearchTimestamp = endFillTime;
+    }
+    if (m_endFillMode || ongoingFill) {
+      getDipData(oms, startSampleTime, endSampleTime);
+      getLumiData(oms, lhcFill, startSampleTime, endSampleTime);
+      if (!m_tmpBuffer.empty()) {
+        boost::posix_time::ptime flumiStart = cond::time::to_boost(m_tmpBuffer.front().first);
+        boost::posix_time::ptime flumiStop = cond::time::to_boost(m_tmpBuffer.back().first);
+        edm::LogInfo(m_name) << "First lumi starts at " << flumiStart << " last lumi starts at " << flumiStop;
+        session.transaction().start(true);
+        getCTPPSData(session, startSampleTime, endSampleTime);
+        session.transaction().commit();
+        session2.transaction().start(true);
+        getEcalData(session2, startSampleTime, endSampleTime);
+        session2.transaction().commit();
+      }
+    }
+
+    if (!m_endFillMode) {
+      if (m_tmpBuffer.size() > 1) {
+        throw cms::Exception("LHCInfoPerFillPopConSourceHandler")
+            << "More than 1 payload buffered for writing in duringFill mode.\
+          In this mode only up to 1 payload can be written";
+      } else if (m_tmpBuffer.size() == 1) {
+        if (theLHCInfoPerFillImpl::comparePayloads(*(m_tmpBuffer.begin()->second), *m_prevPayload)) {
+          m_tmpBuffer.clear();
+          edm::LogInfo(m_name)
+              << "The buffered payload has the same data as the previous payload in the tag. It will not be written.";
+        }
+      } else if (m_tmpBuffer.empty()) {
+        addEmptyPayload(
+            cond::lhcInfoHelper::getFillLastLumiIOV(oms, lhcFill));  //the IOV doesn't matter when using OnlinePopCon
+      }
+      // In duringFill mode, convert the timestamp-type IOVs to lumiid-type IOVs
+      // before transferring the payloads from the buffer to the final collection
+      convertBufferedIovsToLumiid(m_timestampToLumiid);
+    }
+
+    size_t niovs = theLHCInfoPerFillImpl::transferPayloads(m_tmpBuffer, m_iovs, m_prevPayload);
+    edm::LogInfo(m_name) << "Added " << niovs << " iovs within the Fill time";
+    m_tmpBuffer.clear();
+    m_timestampToLumiid.clear();
+
+    if (!m_endFillMode) {
+      return;
+    }
+
+    // endFill mode only:
+    if (m_prevPayload->fillNumber() and !ongoingFill) {
+      if (m_endFillMode) {
+        addEmptyPayload(endFillTime);
+      } else {
+        addEmptyPayload(cond::lhcInfoHelper::getFillLastLumiIOV(oms, lhcFill));
+      }
+    }
+  }
+}
+
+std::string LHCInfoPerFillPopConSourceHandler::id() const { return m_name; }
+
+static constexpr unsigned int kLumisectionsQueryLimit = 4000;
```

Comment:
It is already in the header file
```suggestion
```
But it would be even better if you remove this `constexpr` also from all the header files and use it in the cc's via `cond::lhcInfoHelper::kLumisectionsQueryLimit`, thus centralizing the definition (you are including the header of that helper anyhow)

Reply:
done, I've centralized the definition